src/Visitor/DetectorVisitor.php

```diff
@@ -59,6 +59,17 @@ public function enterNode(Node $node): ?int
             foreach ($this->option->getExtensions() as $extension) {
                 $extension->setOption($this->option);
                 if ($extension->extend($node)) {
+                    if ((
+                        $scalar instanceof LNumber &&
+                        $scalar->getAttribute('kind') != LNumber::KIND_DEC
+                    )) {
+                        $scalar->value = (int) base_convert(
```

Comment:
in this case we are loosing base prefix (0 or 0b). Now dec 10 and binary 10 is displayed in the same way

Reply:
i would like more tests regarding this.